packages/babel-parser/src/plugins/jsx/index.js

```diff
@@ -327,8 +327,22 @@ export default (superClass: Class<Parser>): Class<Parser> =>
         node.expression = this.jsxParseEmptyExpression();
       } else {
         node.expression = this.parseExpression();
+
+        if (
+          node.expression.type === "SequenceExpression" &&
+          ((node.expression.extra &&
+            node.expression.extra.parenthesized === false) ||
+            node.expression.extra === undefined)
```

Comment:
You could do this here, which is a little bit simpler:

```
node.expression.type === "SequenceExpression" &&
          (!node.expression.extra ||
            node.expression.extra.parenthesized !== true)
```
I also used `!== true`, as extra might be an object but parenthesized could be undefined.